### PR TITLE
tab component

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers implements the tab nav (for inside components)

https://zpl.io/2hBKb7

Zeplin: 
screen shot 2016-04-14 at 5 24 01 pm

screen shot 2016-04-14 at 5 23 24 pm
screen shot 2016-04-14 at 5 23 30 pm

### CODE REVIEW

File: shared/more/style-sheet.render.native.js [hunk 1]
```diff
@@ -282,10 +316,19 @@ export default class Render extends Component {
     this.setState({check: next})
   }
 
+  _selectTab (idx: number) {
+    this.setState({tabSelected: [idx === 0, idx === 1, idx === 2]})
```
Comment (cjb) [inline shared/more/style-sheet.render.native.js]: Does this limit a TabBars to three tabs?  (Maybe it's fine if it does, I'm not planning to use it for more.)

Comment (chrisnojima) [inline shared/more/style-sheet.render.native.js]: no, this is the dumb component, just showing what you can do. just add more TabBar.Items 
in practice this doens't horizontally scroll or anything so we'll have to see how it'll have to evolve. right now i think we only ever have 2 tabs. public/private folders and switching input modes for provisioning. it takes a tabWidth but maybe  it should take all the width? we can change it later

File: shared/common-adapters/tab-bar.native.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      if (found) {
+        return
+      }
+
+      if (item.props.selected) {
+        found = item
+      }
+    })
+
+    return found
```
Comment (MarcoPolo) [inline shared/common-adapters/tab-bar.native.js]: replace with `return (children || []).find(i => i.props.selected)` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)

File: shared/common-adapters/tab-bar.native.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          {this._labels()}
+        </Box>
+        {this._content()}
+      </Box>
+    )
+  }
+}
+
+// $FlowIssue dunno why flow doens't like this
+TabBar.Item = TabBarItem
```
Comment (MarcoPolo) [inline shared/common-adapters/tab-bar.native.js]: What's the error you get? It might be because you want to say the type of Item is `Class<TabBarItem>`

**Comment by chrisnojima** [comment]
tnx @MarcoPolo , keep forgetting the class thing

File: shared/common-adapters/tab-bar.native.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+class TabBar extends Component {
+  props: Props;
+  static Item: Class<TabBarItem>;
+
+  _labels (): Array<React$Element> {
+    const tabWidth = this.props.tabWidth || 93
+
+    return (this.props.children || []).map(item => (
```
Comment (cjb) [inline shared/common-adapters/tab-bar.native.js]: `return [].map()` will generate a `ReactElement.create type must not be undefined` warning in RN, so I think you might have to do something like:
```js
    if (!this.props.children) {
      return [<Box/>]
    }
    return (this.props.children).map(item => (
```

Comment (chrisnojima) [inline shared/common-adapters/tab-bar.native.js]: mmm, lemme see, we should be able to return nulls, we shoudln't need empty boxes

Comment (chrisnojima) [inline shared/common-adapters/tab-bar.native.js]: you got this by having no children ?

**Comment by MarcoPolo** [comment]
:+1: not sure about the tab width instead of just having it expand

**Comment by MarcoPolo** [comment]
(but we can change later)

**Comment by chrisnojima** [comment]
in the current design they're fixed width and don't actually take the whole container. but we could limit the size based on the container. i say we move this in and fix up the component as we need it

**Pull request merged**